src/scroll-timeline-base.js

```diff
@@ -181,50 +227,36 @@ function updateSource(timeline, source) {
     let details = sourceDetails.get(source);
     if (!details) {
       // This is the first timeline for this source
-      // Store a list of connected timelines and current scroll position
-      details = {
-        timelines: [],
-        scrollPos: {
-          scrollLeft: source.scrollLeft,
-          scrollTop: source.scrollTop
-        }
-      };
+      // Store a list of connected timelines and current measurements
+      details = {timelines: [], measurements: measureSource(source)};
       sourceDetails.set(source, details);
 
-      const resizeObserver = new ResizeObserver(() => {
-        // Sample and store scroll pos
-        details.scrollPos = {
-          scrollLeft: source.scrollLeft,
-          scrollTop: source.scrollTop
-        };
-        requestAnimationFrame(() => {
-          // Defer ticking timeline to animation frame to prevent
-          // "ResizeObserver loop completed with undelivered notifications"
-          for (const timeline of details.timelines) {
-            renormalizeAnimationTimings(timeline);
-          }
-        });
-      });
+      // Use resize observer to detect changes to source size
+      const resizeObserver = new ResizeObserver(() => updateSourceMeasurements(source));
       resizeObserver.observe(source);
 
+      // Use mutation observer to detect updated style and class attributes on source element
+      const mutationObserver = new MutationObserver(() => updateSourceMeasurements(source));
+      mutationObserver.observe(source, {attributes: true, attributeFilter: ['style', 'class']});
```

Comment:
Sorry to go back and forth on this - I think the ResizeObserver observing each child is better than observing changes to style / class attributes. It would be good to also have a mutationobserver observe added / removed children but this can be done separately.

Reply:
Don't be, I'm truly happy for all the input. 

We can definitely observe added/removed children. I think now is a good time to include that, but I can file a separate PR for that if you'd like that. 


About observing the style attribute:

There are a few tests that toggle whether the timelines are active through toggling the overflow style property.
- https://github.com/web-platform-tests/wpt/blob/55a2476161170898d9f19db51918353b2e2e9640/scroll-animations/scroll-timelines/scroll-animation-inactive-timeline.html#L61
- https://github.com/web-platform-tests/wpt/blob/55a2476161170898d9f19db51918353b2e2e9640/scroll-animations/scroll-timelines/scroll-animation-inactive-timeline.html#L89
 
There were regressions on a few of these tests, as we now have to wait for a startTime/holdTime before committing a pending play task.
 
I added the mutation observer to detect when the overflow property changed through inline styles. The approach is admittedly naive and triggers unnecessary updates, while only detecting a limited subset of the style changes we're interested in.

We can:
1. drop the observer and live with the few regressions, or
2. come up with a smarter way to detect the changes 🤔 Let me know if you have any good ideas 😀

Reply:
> I can file a separate PR for that if you'd like that.

No need! I was worried about properly cleaning up listeners when children are removed / observing new children but the former is handled by the disconnect call, and the latter we can make part of the mutationobserver later.

> I added the mutation observer to detect when the overflow property changed through inline styles. The approach is admittedly naive and triggers unnecessary updates, while only detecting a limited subset of the style changes we're interested in.

Ah I understand, thanks for the explanation. I'm okay with keeping this in for now. For these things which are for handling edge cases correctly for tests but not required most of the time I'd like to eventually put them all behind some flag. E.g. if people are using this in production scenarios they would probably prefer not to pay any additional costs and instead carefully avoid those edge cases or manually trigger an update themselves.

Reply:
Using a flag to toggle this would be nice. 

I'll keep observation of the "style" attribute then, but remove observation of the "class" attribute, as that's not necessary for passing the tests.